components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90

```diff
@@ -3584,334 +3576,608 @@ end subroutine mam_soaexch_1subarea
+                   dryvol_c(imode), deldryvol_c(imode)) !output
+           end if !iscldy then
+        end do
+
+      end subroutine compute_dryvol_change_in_src_mode
+
+      !----------------------------------------------------------------------
+      !----------------------------------------------------------------------
+
+      subroutine dryvolume_change (imode, nspec, q_vmr, q_del_growth, &!input
+           dryvol, deldryvol) !output
+
+        !intent-ins
+        integer,  intent(in) :: imode           !current mode number
+        integer,  intent(in) :: nspec           !number of species in the current mode
+        real(r8), intent(in) :: q_vmr(:,:)        !volume mixing ratio [kmol/kmol] FIXME: units needs to be reverified
+        real(r8), intent(in) :: q_del_growth(:,:) !change (delta) in volume mixing ratio [kmol/kmol]
+
+        !intent-outs
+        real(r8), intent(out) :: dryvol, deldryvol !dry volume (before growth) and its grwoth [m3/kmol]
+
+        !local variables
+        integer  :: ispec, s_spec_ind, e_spec_ind
+        real(r8) :: tmp_dryvol, tmp_del_dryvol
+
+        !For each mode, we compute a dry volume by combining (accumulating) mass/density for each specie in that mode.
+        !conversion from mass to volume is accomplished by multiplying with precomputed "mass_2_vol" factor
+
+        s_spec_ind = 1     !start specie index for this mode [These will be subroutine args]
+        e_spec_ind = nspec !end specie index for this mode
+
+        !initialize tmp accumulators
+        tmp_dryvol     = 0.0_r8 !dry volume accumulator
+        tmp_del_dryvol = 0.0_r8 !dry volume growth(change) accumulator
+
+        !Notes on mass_2_vol factor:Units:[m3/kmol-of-specie]; where kmol-of-specie is the amount of a specie "s"
+        ! This factor is obtained by  (molecular_weight/density) of a specie. That is,
+        ! [ (kg/kmol-of-specie) / (kg/m3) ]; where molecular_weight has units [kg/kmol-of-specie] and density units are [kg/m3]
+        ! which results in the units of m3/kmol-of-specie
+
+        do ispec = s_spec_ind, e_spec_ind
+           !Multiply by mass_2_vol[m3/kmol-of-specie] to convert q_vmr[kmol-of-specie/kmol-air]) to volume units[m3/kmol-air]
+           tmp_dryvol     = tmp_dryvol     + q_vmr(ispec,imode)*mass_2_vol(ispec)        !compute current dryvolume
+           !accumulate the "grwoth" in volume units as well
+           tmp_del_dryvol = tmp_del_dryvol + q_del_growth(ispec,imode)*mass_2_vol(ispec) !compute dryvolume growth
+        end do
+
+        dryvol    = tmp_dryvol-tmp_del_dryvol ! This is dry volume before the growth
+        deldryvol = tmp_del_dryvol          ! change in dry volume due to growth
+
+      end subroutine dryvolume_change
+
+      !----------------------------------------------------------------------
+      !----------------------------------------------------------------------
+
+      subroutine inter_mode_transfer(nmode, nspec, dest_mode_of_mode, &
```

Comment:
Here's another "noun" subroutine name that could be "verbed": `compute_inter_mode_transfer`